src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,142 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+// Note: Some systems need both string.h and strings.h (Savage). However,
+// some systems can't handle both, only use string.h in that case.
+#include <string.h>
+#if defined(HAVE_STRINGS_H) && !defined(NO_STRINGS_WITH_STRING_H)
+#  include <strings.h>
+#endif
+
+
 #ifdef WIN32
-# include "nvim/os/win_defs.h"
+#  include "nvim/os/win_defs.h"
+#else
+#  include "nvim/os/unix_defs.h"
+#endif
+
+
+// on some systems time.h should not be included together with sys/time.h
+#if !defined(HAVE_SYS_TIME_H) || defined(TIME_WITH_SYS_TIME)
+#  include <time.h>
+#endif
+
+#ifdef HAVE_SYS_TIME_H
+#  include <sys/time.h>
+#endif
+
+// Valid on windows, but not recommended
+#include <signal.h>
+
+// Note: if MAXNAMLEN has the wrong value, you will get error messages
+// for not being able to open the swap file.
+#ifndef MAXNAMLEN
+#  ifdef NAME_MAX
+#    define MAXNAMLEN NAME_MAX
+#  else
+     // Seems to be the most supported value.
+#    define MAXNAMLEN 255
+#  endif
+#endif
+
+// Use the system path length if it makes sense.
+#if defined(PATH_MAX) && (PATH_MAX > 1000)
+#  define MAXPATHL       PATH_MAX
 #else
-# include "nvim/os/unix_defs.h"
+#  define MAXPATHL       1024
+#endif
+
+#define BASENAMELEN     (MAXNAMLEN - 5)
+
+#ifndef SYS_VIMRC_FILE
+#  define SYS_VIMRC_FILE "$VIM/nvimrc"
+#endif
+
+#ifndef DFLT_HELPFILE
+#  define DFLT_HELPFILE  "$VIMRUNTIME/doc/help.txt"
+#endif
+
+#ifndef FILETYPE_FILE
+#  define FILETYPE_FILE  "filetype.vim"
+#endif
+
+#ifndef FTPLUGIN_FILE
+#  define FTPLUGIN_FILE  "ftplugin.vim"
+#endif
+
+#ifndef INDENT_FILE
+#  define INDENT_FILE    "indent.vim"
+#endif
+
+#ifndef FTOFF_FILE
+#  define FTOFF_FILE     "ftoff.vim"
 #endif
 
+#ifndef FTPLUGOF_FILE
+#  define FTPLUGOF_FILE  "ftplugof.vim"
+#endif
+
+#ifndef INDOFF_FILE
+#  define INDOFF_FILE    "indoff.vim"
+#endif
+
+#ifndef SYS_MENU_FILE
+#  define SYS_MENU_FILE  "$VIMRUNTIME/menu.vim"
+#endif
+
+#define DFLT_ERRORFILE          "errors.err"
+
+#ifndef SYNTAX_FNAME
+#  define SYNTAX_FNAME   "$VIMRUNTIME/syntax/%s.vim"
+#endif
+
+// Special wildcards that need to be handled by the shell
+#define SPECIAL_WILDCHAR    "`'{"
+
+// size of the command processing buffer
+#define CMDBUFFSIZE 1024
+
+// use up to 5 Mbyte for a buffer
+#ifndef DFLT_MAXMEM
+#  define DFLT_MAXMEM   (5*1024)
+#endif
+
+// use up to 10 Mbyte for Vim
+#ifndef DFLT_MAXMEMTOT
+#  define DFLT_MAXMEMTOT        (10*1024)
+#endif
+
+#if !defined(S_ISDIR) && defined(S_IFDIR)
+#  define S_ISDIR(m) (((m) & S_IFMT) == S_IFDIR)
+#endif
+
+#if !defined(S_ISREG) && defined(S_IFREG)
+#  define S_ISREG(m) (((m) & S_IFMT) == S_IFREG)
+#endif
+
+#if !defined(S_ISBLK) && defined(S_IFBLK)
+#  define S_ISBLK(m) (((m) & S_IFMT) == S_IFBLK)
+#endif
```

Comment:
Is this intended to make `S_ISBLK` work on Windows? I did not think `S_IFMT` was available there.